File: ObjcShowcase/ViewController.m
//
//  ViewController.m
//  ObjcShowcase
//
//  Created by Mikhail Igonin on 23.07.2018.
//  Copyright © 2018 AppSell. All rights reserved.
//

#import "ViewController.h"
#import <AppSellSDK/AppSellSDK.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)initSDK:(id)sender {
    [AppSellSDK initAppWith:@"e7e600ed-54f5-45a6-95b4-726959e8e368" completion:^(NSError * _Nullable error) {
        [self showMessageForError:error];
    }];
}

- (IBAction)sendEvent:(id)sender {
    [AppSellSDK sendEvent:@"registration" completion:^(NSError * _Nullable error) {
        [self showMessageForError:error];
    }];
}

- (IBAction)getReferrerID:(id)sender {
    NSString* referrerID = [AppSellSDK referrerID];
    NSString* message = nil;
    
    if (referrerID == nil) {
        message = @"ReferrerID = nil";
    } else {
        message = [NSString stringWithFormat:@"ReferrerID = %@", referrerID];
    }
    
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"" message:message preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}
    
- (void)showMessageForError: (nullable NSError*)error {
    NSString* message = nil;
    
    if (error == nil) {
        message = @"Success";
    } else {
        message = [NSString stringWithFormat:@"Error: %@", error.localizedDescription];
    }
    
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"" message:message preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}

@end

File: SwiftShowcase/ViewController.swift
//
//  ViewController.swift
//  DeepLinksTest
//
//  Created by Mikhail Igonin on 08.07.2018.
//  Copyright © 2018 Mikhail Igonin. All rights reserved.
//

import UIKit
import AppSellSDK

class ViewController: UIViewController {

    let appID = "e7e600ed-54f5-45a6-95b4-726959e8e368"
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
     
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func initSDK() {
        AppSellSDK.initApp(with: appID) { (error: Error?) in
            //Check for error
            self.showMessage(for: error)
        }
    }
    
    @IBAction func sendEvent() {
        AppSellSDK.sendEvent("registration") { (error) in
            self.showMessage(for: error)
        }
    }
    
    @IBAction func getReferrerID() {
        let referrerID = AppSellSDK.referrerID
        let message = "ReferrerID = \(referrerID != nil ? referrerID! : "nil")"
        
        let alert = UIAlertController(title: "", message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
        
        present(alert, animated: true, completion: nil)

    }

    func showMessage(for error: Error?) {
        let message: String

        
        if error == nil {
            message = "Success"
        } else {
            message = "Error: " + error!.localizedDescription
        }
        
        let alert = UIAlertController(title: "", message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
        
        present(alert, animated: true, completion: nil)
    }
}

